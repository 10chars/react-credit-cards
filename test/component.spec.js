import React from 'react';
import { mount } from 'enzyme';

import Cards from 'index';

const props = {
  name: 'John Smith',
  number: '4901090345347688',
  expiry: '12/18',
  cvc: '457',
  focused: '',
};

function setup(ownProps = props) {
  return mount(<Cards {...ownProps} />);
}

describe('Cards', () => {
  const wrapper = setup();

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
