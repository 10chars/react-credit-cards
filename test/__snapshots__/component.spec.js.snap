exports[`Cards should render properly 1`] = `"<div class=\"rccs\"><div class=\"rccs__card rccs__card--visa\"><div class=\"rccs__card--front\"><div class=\"rccs__issuer\"></div><div class=\"rccs__cvc--front\">457</div><div class=\"rccs__number\">4901 0903 4534 7688</div><div class=\"rccs__name\">John Smith</div><div class=\"rccs__expiry\"><div class=\"rccs__expiry__valid\">valid thru</div><div class=\"rccs__expiry__value\">12/18</div></div><div class=\"rccs__chip\"></div></div><div class=\"rccs__card--back\"><div class=\"rccs__stripe\"></div><div class=\"rccs__signature\"></div><div class=\"rccs__cvc\">457</div><div class=\"rccs__issuer\"></div></div></div></div>"`;
